### PR TITLE
separate the run, run-upgrade, and run-test

### PR DESCRIPTION
**Pull request by deads2k**
/hold

opening so I don't lose it, but this won't work until the core of the invariant checking lands.  Darn you images!

### CODE REVIEW

File: pkg/cmd/openshift-tests/images/images_command.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	Verify     bool
+}
+
+// createImageMirrorForInternalImages returns a list of 'oc image mirror' mappings from source to
+// target or returns an error. If mirrored is true the images are assumed to have already been copied
+// from their upstream location into our official mirror, in the REPO:TAG format where TAG is a hash
+// of the original internal name and the index of the image in the array. Otherwise the mappings will
+// be set to mirror the location as defined in the test code into our official mirror, where the target
+// TAG is the hash described above.
+func createImageMirrorForInternalImages(prefix string, ref reference.DockerImageReference, mirrored bool) ([]string, error) {
```
Comment (deads2k) [inline pkg/cmd/openshift-tests/images/images_command.go:108]: moved, not new

File: pkg/cmd/openshift-tests/run-upgrade/command.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+
+	"github.com/openshift/origin/pkg/clioptions/imagesetup"
+	"github.com/openshift/origin/pkg/testsuites"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+func NewRunUpgradeCommand(streams genericclioptions.IOStreams) *cobra.Command {
```
Comment (deads2k) [inline pkg/cmd/openshift-tests/run-upgrade/command.go:14]: moved, not new

File: pkg/cmd/openshift-tests/run/command.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	"fmt"
+
+	"github.com/openshift/origin/pkg/clioptions/imagesetup"
+	"github.com/openshift/origin/pkg/testsuites"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+func NewRunCommand(streams genericclioptions.IOStreams) *cobra.Command {
```
Comment (deads2k) [inline pkg/cmd/openshift-tests/run/command.go:14]: moved, not new

File: pkg/cmd/openshift-tests/run/flags.go [hunk 4]
```diff
@@ -82,7 +81,6 @@ func (f *RunSuiteFlags) ToOptions(args []string) (*RunSuiteOptions, error) {
 	if err != nil {
 		return nil, err
 	}
-	ginkgoOptions.SyntheticEventTests = pulledInvalidImages(f.FromRepository)
```
Comment (deads2k) [inline pkg/cmd/openshift-tests/run/flags.go:85]: moved to invariant

**Comment by deads2k** [comment]
/hold cancel

prereqs merged.

**Comment by stbenjam** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade 10

**Comment by stbenjam** [comment]
Feel free to cancel hold after aggregation passes.  This looks mostly straight forward but since that images test was touched I want to be sure I'm not going to ruin Devan's Friday with another regression

**Comment by deads2k** [comment]
/payload-aggregate periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade 10

**Comment by deads2k** [comment]
aggregation passed.  no conflict with existing.  merging

https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/aggregator-periodic-ci-openshift-release-master-ci-4.14-upgrade-from-stable-4.13-e2e-aws-ovn-upgrade/1687116752703459328

**Pull request merged**